game.py

```diff
@@ -20,8 +20,35 @@ def snowman(snowman_word):
     If the player wins and, 
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
-
+    # build a dict to store letter and status (T or F) for each letter in word
+    letter_statuses_dict = build_letter_status_dict(snowman_word)
+    # define a wrong guesses list
+    wrong_guesses_list = []
+
+    # loop condition: wrong guesses < MAX wrong guesses
+    while len(wrong_guesses_list) < SNOWMAN_MAX_WRONG_GUESSES:
+        # get user input
+        user_input = get_letter_from_user(letter_statuses_dict, wrong_guesses_list)
+        # check if user_input is in the word
+        # if yes, change status of the letter in the letter_statuses_dict
+        if user_input in letter_statuses_dict:
+            letter_statuses_dict[user_input] = True
+            # check is_word_guessed, if yes, player wins, print message, exit;
+            if is_word_guessed(snowman_word, letter_statuses_dict):
```

Comment:
While this isn't inherently too big of a deal, it can be a good idea to avoid nested conditionals if it isn't necessary. In this particular instance, checking if the word has been guessed correctly isn't dependent upon whether or not the guessed letter is in the word. If we moved this nested if statement to be a standalone conditional, the code will work the exact same way. In a situation like this where the location of the if statement does not change how the code works, it's generally a good idea to err on the side of un-nested conditionals!